main/lsp/ErrorReporter.h

```diff
@@ -0,0 +1,32 @@
+#ifndef RUBY_TYPER_LSP_ERRORREPORTER_H
+#define RUBY_TYPER_LSP_ERRORREPORTER_H
+
+#include "ast/ast.h"
+#include "core/core.h"
+#include "main/lsp/LSPConfiguration.h"
+
+namespace sorbet::realmain::lsp {
+
+class ErrorReporter {
+    const std::shared_ptr<const LSPConfiguration> config;
+    /** Stores the epoch in which we last sent diagnostics to the client for each file. */
+    std::vector<u4> diagnosticEpochs;
+    /** List of files that have had errors in last run*/
+    UnorderedSet<core::FileRef> filesThatHaveErrors;
```

Comment:
seems like your life will be easier if you were to replace this vector&set with a map.

Reply:
Good call. Since `diagnosticEpochs` should contain one entry per file, we can use a vector like a map. Maybe something like:

```
struct ErrorStatus {
  // The epoch at which we last sent diagnostics for this file.
  u4 sentEpoch;
  // If true, the client believes this file has errors.
  bool hasErrors;
};

// Maps from file ref ID to its error status.
std::vector<ErrorStatus> fileErrorStatus;
```